ci-operator/step-registry/openstack/scale/masters/openstack-scale-masters-commands.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+info() {
+	printf '%s: %s\n' "$(date --utc +%Y-%m-%dT%H:%M:%SZ)" "$*"
+}
+
+if [ -z "$SCALE_MASTERS_FLAVOR_ALTERNATE" ]; then
+	info "All defined environment variables for the step are empty. Exiting."
+	exit 0
+fi
+
+declare patch=''
+
+info "Detected the environment variable SCALE_MASTERS_FLAVOR_ALTERNATE set to '$SCALE_MASTERS_FLAVOR_ALTERNATE'. Patching the CPMS..."
+
+patch="$(printf '{"%s": "%s"}' 'spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.flavor' "$(<"${SHARED_DIR}/OPENSTACK_CONTROLPLANE_FLAVOR_ALTERNATE")")"
+
+oc patch controlplanemachineset.machine.openshift.io --namespace openshift-machine-api cluster -p "$patch"
+
+sleep 5
+
+info 'Waiting for the rollout to complete...'
+# shellcheck disable=SC2046
+oc wait --timeout=90m --for=condition=Progressing=false cluster-control-plane-machine-set-operator -o template='{{.metadata.name}} is ready
+'
+
```

Comment:
Here I think we want to add some validations:
* (using `oc` client) Is the `providerSpec` updated for the machines?
* (using `openstack` client) Are the new machines using the right flavour?

Reply:
As these you mention really are tests, I'd add them to [our `openshift/openstack` tests](https://github.com/openshift/openstack-test/tree/main/test/extended/openstack):

IF CPMS is enabled:
* test that the flavor in the Control plane machines matches the one in the Machines
* test that the flavor in the Machines matches the one in the instances (we sort of already have this one)